airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -230,7 +240,22 @@ public SourceDiscoverSchemaRead discoverSchemaForSourceFromSourceId(final Source
     if (currentCatalog.isEmpty() || bustActorCatalogCache) {
       final SynchronousResponse<UUID> persistedCatalogId =
           synchronousSchedulerClient.createDiscoverSchemaJob(source, imageName, connectorVersion, new Version(sourceDef.getProtocolVersion()));
-      return retrieveDiscoveredSchema(persistedCatalogId);
+      final SourceDiscoverSchemaRead discoveredSchema = retrieveDiscoveredSchema(persistedCatalogId);
+
+      if (discoverSchemaRequestBody.getConnectionId() != null) {
```

Comment:
When can this be null. Should we log an error if so?

Reply:
I believe this endpoint is called when connections are created in order to discover schemas, and in this case we don't need to pass in the connection id since there is not an existing sync catalog to compare anything against